core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -318,24 +319,23 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
         (new InMemoryStore(), true)
```

Comment:
not related to this change, comment is really for `val _replay = !path.isDirectory()` but github won't let me put comment there ...

I know we discussed this when you made this change, but still I was confused reading this bit of code on replay.  Maybe could you just add a comment above that line like "the kvstore is deleted when we decide that the loaded data is stale -- see LoadedAppUI for a more extensive discussion of the lifecycle".